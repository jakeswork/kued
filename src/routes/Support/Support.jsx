import React, { Component } from 'react';
import {
  FiAlertTriangle,
  FiHeart,
  FiThumbsUp,
  FiMessageCircle,
} from 'react-icons/fi';

import propTypes from './types';
import MainLayout from '../../layouts/MainLayout';
import Text from '../../components/Text';
import Card from '../../components/Card';
import Emoji from '../../components/Emoji';
import ResponsiveModal from './components/ResponsiveModal';
import Button from '../../components/Button';

export const cardStyles = {
  cursor: 'pointer',
  minWidth: 240,
};

class Support extends Component {
  static propTypes = propTypes;

  state = {
    showType: null,
  }

  close = cb => this.setState(({ showType: null }), typeof cb === 'function' ? cb : () => {})

  showHelp = showType => this.close(() => setTimeout(() => this.setState({ showType }), 300))

  render() {
    const { classes } = this.props;
    const { showType } = this.state;

    return (
      <MainLayout>
        <Text h1>
          How can we help?
        </Text>
        <Text h5>
          Kued was developed to help level the playing field.
          By providing an open view of everything that is going on within your server,
          you can make better decisions on what, where and when you play.
        </Text>
        <Text caption>Click on a card to learn more.</Text>
        <div className={classes.cardsWrapper}>
          <Card
            style={cardStyles}
            onClick={() => this.showHelp('support')}
            data-test-id="supportCard"
          >
            <Text h4>
              Help us help you&nbsp;
              <Emoji icon="🤝" label="handshake" />
            </Text>
            <Text bold caption>Want to help out by supporting the project?</Text>
          </Card>
          <Card style={cardStyles} onClick={() => this.showHelp('bug')}>
            <Text h4>
              Something not right?&nbsp;
              <Emoji icon="🛠" label="hammer-and-wrench" />
            </Text>
            <Text bold caption>Let us know if you&apos;re having any problems.</Text>
          </Card>
          <Card style={cardStyles} onClick={() => this.showHelp('vote')}>
            <Text h4>
              Missing out?&nbsp;
              <Emoji icon="😭" label="loudly-crying-face" />
            </Text>
            <Text bold caption>Vote for us to come to your server.</Text>
          </Card>
        </div>
        <ResponsiveModal in={showType === 'support'} onClose={this.close}>
          <Text h3>Supporting Kued</Text>
          <Text>
            Kued is an independent project with ongoing development,
            made possible thanks to the support of awesome backers.
            If you&apos;d like to help us out, please consider joining them!
          </Text>
          <Button
            icon={<FiHeart />}
            success
            analyticsLabel="donationLink"
            link="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=JG5ENWX7ADSKL&source=url"
          >
            Donate via PayPal
          </Button>
        </ResponsiveModal>
        <ResponsiveModal in={showType === 'bug'} onClose={this.close}>
          <Text h3>Reporting Issues</Text>
          <Text>
            We want to make Kued the best it can be. To do so, we need your help.
            We call on all users of Kued to report any problems
            they have encountered so that we can tackle them head on
            and give all of our users the best experience.
          </Text>
          <Button
            danger
            link="https://forms.gle/gu9fpS2RZytgzSbj9"
            icon={<FiAlertTriangle />}
            analyticsLabel="bugReportLink"
          >
            Report a bug
          </Button>
        </ResponsiveModal>
        <ResponsiveModal in={showType === 'vote'} onClose={this.close}>
          <Text h3>Server Requests</Text>
          <Text>
            Is your server not yet supported by Kued?
            Let your voice be heard! Head over and vote for your server
            to let us know where we&apos;re wanted.
          </Text>
          <Text bold caption>
            Once a server reaches 50 vote points it will be considered for development.
          </Text>
          <Button link="/servers" icon={<FiThumbsUp />}>Vote for your server</Button>
        </ResponsiveModal>
        <div className={classes.contactFooter}>
          <Text bold>
            You can click below to talk to somebody directly.
          </Text>
          <Button
            link="https://www.twitch.tv/apparentt/"
            icon={<FiMessageCircle />}
            analyticsLabel="twitchMessageLink"
          >
            Contact
          </Button>
        </div>
      </MainLayout>
    );
  }
}

export default Support;
