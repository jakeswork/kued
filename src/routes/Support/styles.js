export default theme => ({
  cardsWrapper: {
    display: 'flex',
    justifyContent: 'center',
    flexWrap: 'wrap',
  },
  contactFooter: {
    marginTop: 64,
  },
  [theme.media.mobile]: {
    contactFooter: {
      marginTop: 32,
    },
  },
});
