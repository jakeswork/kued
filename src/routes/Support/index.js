import withStyles from 'react-jss';

import Support from './Support';
import styles from './styles';

export default withStyles(styles)(Support);
