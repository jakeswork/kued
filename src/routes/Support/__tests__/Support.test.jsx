import React from 'react';
import { shallow } from 'enzyme';

import Support, { cardStyles } from '../Support';

const defaultProps = {
  classes: {},
};

describe('The Support instance', () => {
  describe('When called with default properties', () => {
    it('should render without throwing any errors', () => {
      const wrapper = shallow(<Support {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when clicking on the cards', () => {
    it('should fire the showHelp method', () => {
      jest.useFakeTimers();

      const wrapper = shallow(<Support {...defaultProps} />);
      const cards = wrapper.find({ style: cardStyles });

      cards.forEach((card, i) => {
        card.simulate('click');

        jest.advanceTimersByTime(300);

        if (i === 0) return expect(wrapper.state('showType')).toEqual('support');

        if (i === 1) return expect(wrapper.state('showType')).toEqual('bug');

        if (i === 2) return expect(wrapper.state('showType')).toEqual('vote');
      });
    });
  });

  describe('when calling the close method', () => {
    it('should set the state to null', () => {
      const wrapper = shallow(<Support {...defaultProps} />);

      wrapper.setState({ showType: 'test' });

      expect(wrapper.state('showType')).toEqual('test');

      wrapper.instance().close();

      expect(wrapper.state('showType')).toEqual(null);
    });
  });
});
