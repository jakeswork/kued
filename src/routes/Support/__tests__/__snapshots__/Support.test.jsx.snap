// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Support instance When called with default properties should render without throwing any errors 1`] = `
<Jss(MainLayout)>
  <Jss(Text)
    bold={false}
    caption={false}
    h1={true}
    h2={false}
    h3={false}
    h4={false}
    h5={false}
    link={null}
    style={Object {}}
  >
    How can we help?
  </Jss(Text)>
  <Jss(Text)
    bold={false}
    caption={false}
    h1={false}
    h2={false}
    h3={false}
    h4={false}
    h5={true}
    link={null}
    style={Object {}}
  >
    Kued was developed to help level the playing field. By providing an open view of everything that is going on within your server, you can make better decisions on what, where and when you play.
  </Jss(Text)>
  <Jss(Text)
    bold={false}
    caption={true}
    h1={false}
    h2={false}
    h3={false}
    h4={false}
    h5={false}
    link={null}
    style={Object {}}
  >
    Click on a card to learn more.
  </Jss(Text)>
  <div>
    <Jss(Card)
      className={null}
      data-test-id="supportCard"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "minWidth": 240,
        }
      }
    >
      <Jss(Text)
        bold={false}
        caption={false}
        h1={false}
        h2={false}
        h3={false}
        h4={true}
        h5={false}
        link={null}
        style={Object {}}
      >
        Help us help you 
        <Jss(Emoji)
          emoji={null}
          icon="🤝"
          label="handshake"
        />
      </Jss(Text)>
      <Jss(Text)
        bold={true}
        caption={true}
        h1={false}
        h2={false}
        h3={false}
        h4={false}
        h5={false}
        link={null}
        style={Object {}}
      >
        Want to help out by supporting the project?
      </Jss(Text)>
    </Jss(Card)>
    <Jss(Card)
      className={null}
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "minWidth": 240,
        }
      }
    >
      <Jss(Text)
        bold={false}
        caption={false}
        h1={false}
        h2={false}
        h3={false}
        h4={true}
        h5={false}
        link={null}
        style={Object {}}
      >
        Something not right? 
        <Jss(Emoji)
          emoji={null}
          icon="🛠"
          label="hammer-and-wrench"
        />
      </Jss(Text)>
      <Jss(Text)
        bold={true}
        caption={true}
        h1={false}
        h2={false}
        h3={false}
        h4={false}
        h5={false}
        link={null}
        style={Object {}}
      >
        Let us know if you're having any problems.
      </Jss(Text)>
    </Jss(Card)>
    <Jss(Card)
      className={null}
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "minWidth": 240,
        }
      }
    >
      <Jss(Text)
        bold={false}
        caption={false}
        h1={false}
        h2={false}
        h3={false}
        h4={true}
        h5={false}
        link={null}
        style={Object {}}
      >
        Missing out? 
        <Jss(Emoji)
          emoji={null}
          icon="😭"
          label="loudly-crying-face"
        />
      </Jss(Text)>
      <Jss(Text)
        bold={true}
        caption={true}
        h1={false}
        h2={false}
        h3={false}
        h4={false}
        h5={false}
        link={null}
        style={Object {}}
      >
        Vote for us to come to your server.
      </Jss(Text)>
    </Jss(Card)>
  </div>
  <Jss(ResponsiveModal)
    in={false}
    onClose={[Function]}
  >
    <Jss(Text)
      bold={false}
      caption={false}
      h1={false}
      h2={false}
      h3={true}
      h4={false}
      h5={false}
      link={null}
      style={Object {}}
    >
      Supporting Kued
    </Jss(Text)>
    <Jss(Text)
      bold={false}
      caption={false}
      h1={false}
      h2={false}
      h3={false}
      h4={false}
      h5={false}
      link={null}
      style={Object {}}
    >
      Kued is an independent project with ongoing development, made possible thanks to the support of awesome backers. If you'd like to help us out, please consider joining them!
    </Jss(Text)>
    <Jss(Button)
      analyticsLabel="donationLink"
      danger={false}
      flat={false}
      icon={<FiHeart />}
      link="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=JG5ENWX7ADSKL&source=url"
      secondary={false}
      success={true}
    >
      Donate via PayPal
    </Jss(Button)>
  </Jss(ResponsiveModal)>
  <Jss(ResponsiveModal)
    in={false}
    onClose={[Function]}
  >
    <Jss(Text)
      bold={false}
      caption={false}
      h1={false}
      h2={false}
      h3={true}
      h4={false}
      h5={false}
      link={null}
      style={Object {}}
    >
      Reporting Issues
    </Jss(Text)>
    <Jss(Text)
      bold={false}
      caption={false}
      h1={false}
      h2={false}
      h3={false}
      h4={false}
      h5={false}
      link={null}
      style={Object {}}
    >
      We want to make Kued the best it can be. To do so, we need your help. We call on all users of Kued to report any problems they have encountered so that we can tackle them head on and give all of our users the best experience.
    </Jss(Text)>
    <Jss(Button)
      analyticsLabel="bugReportLink"
      danger={true}
      flat={false}
      icon={<FiAlertTriangle />}
      link="https://forms.gle/gu9fpS2RZytgzSbj9"
      secondary={false}
      success={false}
    >
      Report a bug
    </Jss(Button)>
  </Jss(ResponsiveModal)>
  <Jss(ResponsiveModal)
    in={false}
    onClose={[Function]}
  >
    <Jss(Text)
      bold={false}
      caption={false}
      h1={false}
      h2={false}
      h3={true}
      h4={false}
      h5={false}
      link={null}
      style={Object {}}
    >
      Server Requests
    </Jss(Text)>
    <Jss(Text)
      bold={false}
      caption={false}
      h1={false}
      h2={false}
      h3={false}
      h4={false}
      h5={false}
      link={null}
      style={Object {}}
    >
      Is your server not yet supported by Kued? Let your voice be heard! Head over and vote for your server to let us know where we're wanted.
    </Jss(Text)>
    <Jss(Text)
      bold={true}
      caption={true}
      h1={false}
      h2={false}
      h3={false}
      h4={false}
      h5={false}
      link={null}
      style={Object {}}
    >
      Once a server reaches 50 vote points it will be considered for development.
    </Jss(Text)>
    <Jss(Button)
      analyticsLabel={null}
      danger={false}
      flat={false}
      icon={<FiThumbsUp />}
      link="/servers"
      secondary={false}
      success={false}
    >
      Vote for your server
    </Jss(Button)>
  </Jss(ResponsiveModal)>
  <div>
    <Jss(Text)
      bold={true}
      caption={false}
      h1={false}
      h2={false}
      h3={false}
      h4={false}
      h5={false}
      link={null}
      style={Object {}}
    >
      You can click below to talk to somebody directly.
    </Jss(Text)>
    <Jss(Button)
      analyticsLabel="twitchMessageLink"
      danger={false}
      flat={false}
      icon={<FiMessageCircle />}
      link="https://www.twitch.tv/apparentt/"
      secondary={false}
      success={false}
    >
      Contact
    </Jss(Button)>
  </div>
</Jss(MainLayout)>
`;
